### ISSUE TITLE
[Feature Req] A static UI parser based on FTXUI

### ISSUE THREAD

**Issue by SAtacker** [ArthurSonzogni/FTXUI#293]
Hi, Arthur (not tagging you as you are busy at this moment),
I had an idea to make a quick Static UI tool built on top of FTXUI
Features:
- Hooks to make it feel dynamic
- No runtime dependencies (although we can have FTXUI itself as the only one)
- Quick YAML or JSON like file to generate UI

**Comment by ArthurSonzogni** [comment]
Hello Shreyas!

This might be a lovely idea, depending on what this is at the end. Would you have some ideas about how it could look like?

**Comment by SAtacker** [comment]
Basically the idea is to generate some customisable UI (although static or in future hooks for events). An example could be [Whiptail](https://en.m.wikibooks.org/wiki/Bash_Shell_Scripting/Whiptail).

**Comment by ArthurSonzogni** [comment]
So your goal is to avoid compiling C++, but let user's to define some textual UI themselve. The UI would be interpreted.

That's a good thing to have, but the difficulty is to make a language that is nice to use and sufficiently expressive.

**Comment by SAtacker** [comment]
Yes I agree, it's difficult and even more so with event based/ hooks.
We need concepts of Signals for events, State of component and communication between events I guess (context: Qt).

**Comment by omeid** [comment]
Sounds like HTML to me.

**Comment by SAtacker** [comment]
https://github.com/SAtacker/quick-ftxui ongoing, maybe it will complete in Summers next year
